In [47]:
import nltk
import pandas as pd
import os

messages=pd.read_csv(r'C:\Users\abhis\Desktop\dataset.csv', names=["category",'message'])
# print(messages)




In [49]:
#data cleaning and preprocessing
#stemming is faster then lemmetization so here i m used stemming
import re
from nltk.corpus import stopwords
#porterstemmer using for stem its a techinque that find the root word 
# from nltk.stem.porter import PorterStemmer
from nltk.stem import WordNetLemmatizer
# ps=PorterStemmer()
lm=WordNetLemmatizer()
corpus=[]
for i in  range(0,len(messages)):
    r=re.sub('[^a-zA-Z]', ' ',messages['message'][i])
    r=r.lower()
    r=r.split()
    
    r=[lm.lemmatize(word) for word in r if not word in stopwords.words('english')]
    r=' '.join(r)
    corpus.append(r)
# print(corpus)

In [41]:
#using beg of word
from sklearn.feature_extraction.text  import CountVectorizer
cv=CountVectorizer()
X=cv.fit_transform(corpus).toarray()

y=pd.get_dummies(messages['category'])# it used for change ham and spam "1"
print(y)
y=y.iloc[:,1].values #it used for takes one column
print(y)

      Category  ham  spam
0            1    0     0
1            0    1     0
2            0    1     0
3            0    0     1
4            0    1     0
...        ...  ...   ...
5568         0    0     1
5569         0    1     0
5570         0    1     0
5571         0    1     0
5572         0    1     0

[5573 rows x 3 columns]
[0 1 1 ... 1 1 1]


In [42]:
#test train split
from sklearn.model_selection import train_test_split
X_train,X_test,y_train, y_test=train_test_split(X,y,test_size=0.20,random_state=0)
# print(X_train,X_test,y_train, y_test)

In [43]:
# training model using navie bayes classification
from sklearn.naive_bayes import MultinomialNB
spam_detect_model=MultinomialNB().fit(X_train,y_train)

y_pred=spam_detect_model.predict(X_test)
print(y_pred)

from sklearn.metrics import confusion_matrix,accuracy_score
confusion_m=confusion_matrix(y_pred,y_test)
print(confusion_m)

accuracy=accuracy_score(y_pred,y_test)
print(accuracy)


[1 1 1 ... 1 1 1]
[[131  15]
 [  5 964]]
0.9820627802690582


using Stemming i got 0.9820627802690582 accuracy and  i used lemmatizer i got 0.9820627802690582